index.js

```diff
@@ -1,21 +1,35 @@
-// Inspired by (but complete rewrite of) requirejs-metagen
-let _ = require('lodash/fp')
-let Promise = require('bluebird')
-let readDir = Promise.promisify(require('recursive-readdir'))
-let fs = Promise.promisifyAll(require('fs'))
-
-// Path Utils
-let filesRelative = (dir, ex) => readDir(dir, ex).map(x => x.slice(dir.length))
+// Inspired by (but complete rewrite of) requirejs-metagen.
+let _       = require('lodash/fp') // Add lodash helper functions.
+let Promise = require('bluebird')  // Replace promises with the Bluebird Promises library. ("Bluebird cuts corners" -- yikes!)
+let readDir = Promise.promisify(require('recursive-readdir')) // Lists all files in a certain directory and all its sub-directories.
+let fs      = Promise.promisifyAll(require('fs')) // Add the filesystem library.
+
+// Define path Utils.
+
+// Get all the file names in a certain directory and its sub-directories, excluding the directory path.
+let filesRelative = (dir, exclusions) => readDir(dir, exclusions).map(filename => filename.slice(dir.length))
+
+// Get a file's name without its extension.
 let noExt = file => file.slice(0, _.lastIndexOf('.', file))
-let test = x => y => x.test(y)
 
-// Core
-let filter = _.filter(test(/.js|.html|.jsx|.ts|.coffee|.less|.css|.sass|.hbs|.ejs/))
+// Slightly more convenient way to return a function which performs a certain regex match on a string.
+let stringMatch = regex => str => regex.test(str)
```

Comment:
This was intentionally named `test` because it matches the [ramda `test` function](http://ramdajs.com/docs/#test). In fact, this should probably be moved into `futil-js`.